scripts/check-RN-link-perms.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+# Checks if a release notes file is updated. If true, searches the release notes file for links to bugs that are behind a login and fails the build.
+# We should not include internal bugs in our release notes.
+
+# Get the repo path and files changed
+REPO_PATH="$(git rev-parse --show-toplevel)"
+FILES=$(git diff --name-only HEAD~1 HEAD --diff-filter=d "*.adoc" ':(exclude)_unused_topics/*')
+
+# Function to check links in updated release notes
+check_rn_links () {
+    
+    #Extract the release notes file from all the updated files
+    while IFS= read -r line; do
+        if echo "$line" | grep -q "release"; then
+            RELEASE_FILE=$(echo "$line")
+        fi
+    done <<< "$FILES"
+
+    echo ""
+    echo "#########"
+    echo "You updated the following release notes file:"
+    echo "$RELEASE_FILE"
+    echo "#########"
+    echo ""
+
+    # Read the content from the local release note file
+    content=$(cat "$REPO_PATH/$RELEASE_FILE")
+
+    # Extract links from the content
+    links=$(echo "$content" | grep -o 'https://issues[^]]*' | sed 's/\[.*//')
+    protected_links=()
+
+    # Need to discount links in comments.
+
+    echo ""
+    echo "#########"
+    echo "Checking for internal bug links in $REPO_PATH/$RELEASE_FILE"
+    echo "#########"
+    echo ""
+
+    # Iterate over the links and check their authorization status
+    for link in $links
+    do
+        response=$(curl -I "$link" 2>&1)
+
+        if echo "$response" | grep -q "permissionViolation"; then
+            echo "The link $link requires authentication."
+            protected_links+=("$link")
+        else
+            echo "The link $link does not require authentication."
```

Comment:
IMO shouldn't need to flag links that are OK.

Reply:
Agreed.... just in there for testing purposes now